api4/post_test.go

```diff
@@ -1993,6 +1993,54 @@ func TestDeletePost(t *testing.T) {
 	CheckNoError(t, resp)
 }
 
+func TestDeletePostMessage(t *testing.T) {
+	th := Setup(t).InitBasic()
+	th.LinkUserToTeam(th.SystemAdminUser, th.BasicTeam)
+	th.App.AddUserToChannel(th.SystemAdminUser, th.BasicChannel)
+
+	defer th.TearDown()
+
+	wsClient, err := th.CreateWebSocketClientWithClient(th.Client)
+	require.Nil(t, err)
+	defer wsClient.Close()
+
+	wsClient.Listen()
+
+	wsAdminClient, err := th.CreateWebSocketClientWithClient(th.SystemAdminClient)
+	require.Nil(t, err)
+	defer wsAdminClient.Close()
+
+	wsAdminClient.Listen()
+
+	waitForEvent := func(client *model.WebSocketClient, delete_by interface{}) {
```

Comment:
Was `delete_by` made to be `interface{}` because the map value is `interface{}`? Wondering if we could just pass `*string` and then check if value exists or not in the map if it's nil?

Reply:
I originally tried something along these lines but I decided on a more uniform version of the code. I also think it is better to use the "correct" types for the assertion

Reply:
Ok.